pyccl/halos/halo_model.py

```diff
@@ -23,83 +23,68 @@ class HMCalculator(CCLAutoreprObject):
         integration_method_M (string): integration method to use
             in the mass integrals. Options: "simpson" and "spline".
             Default: "simpson".
-        k_min (float): some of the integrals solved by this class
+        k_norm (float): some of the integrals solved by this class
             will often be normalized by their value on very large
             scales. This parameter (in units of inverse Mpc)
             determines what is considered a "very large" scale.
             Default: 1E-5.
     """
-    __repr_attrs__ = ("_massfunc", "_hbias", "_mdef", "_prec",)
-
-    def __init__(self, cosmo, massfunc, hbias, mass_def,
-                 log10M_min=8., log10M_max=16.,
-                 nlog10M=128, integration_method_M='simpson',
-                 k_min=1E-5):
-        # halo mass definition
-        if isinstance(mass_def, MassDef):
-            self._mdef = mass_def
-        elif isinstance(mass_def, str):
-            self._mdef = MassDef.from_name(mass_def)()
-        else:
-            raise TypeError("mass_def must be of type `MassDef` "
-                            "or a mass definition name string")
-
-        # halo mass function
-        if isinstance(massfunc, MassFunc):
-            self._massfunc = massfunc
-        elif isinstance(massfunc, str):
-            nMclass = MassFunc.from_name(massfunc)
-            self._massfunc = nMclass(cosmo, mass_def=self._mdef)
-        else:
-            raise TypeError("mass_function must be of type `MassFunc` "
-                            "or a mass function name string")
-
-        # halo bias function
-        if isinstance(hbias, HaloBias):
-            self._hbias = hbias
-        elif isinstance(hbias, str):
-            bMclass = HaloBias.from_name(hbias)
-            self._hbias = bMclass(cosmo, mass_def=self._mdef)
-        else:
-            raise TypeError("halo_bias must be of type `HaloBias` "
-                            "or a halo bias name string")
-
-        self._prec = {'log10M_min': log10M_min,
-                      'log10M_max': log10M_max,
-                      'nlog10M': nlog10M,
-                      'integration_method_M': integration_method_M,
-                      'k_min': k_min}
-        self._lmass = np.linspace(self._prec['log10M_min'],
-                                  self._prec['log10M_max'],
-                                  self._prec['nlog10M'])
+    __repr_attrs__ = ("mass_function", "halo_bias", "mass_def", "precision",)
+    __getattr__ = deprecate_attr(pairs=[('_mdef', 'mass_def'),
+                                        ('_massfunc', 'mass_function'),
+                                        ('_hbias', 'halo_bias'),
+                                        ('_prec', 'precision')]
+                                 )(super.__getattribute__)
+
+    @warn_api(pairs=[("massfunc", "mass_function"), ("hbias", "halo_bias"),
+                     ("log10M_min", "lM_min"), ("log10M_max", "lM_max"),
```

Comment:
I'm rethinking renaming `log10` -> `l`.
a) log10 is more informative
b) we use `lk_arr` in `Pk2D` and `Tk3D` to label natural logarithms.

Reply:
I tend to agree. Not every name change I made back then was on point.
However, we have to decide what we'll do with e.g. the HOD profile where we use `lMmin, lM0, lM1` etc.

Reply:
Let's consistently use log10 for log10 (or l10 if you prefer, although I find that more difficult to parse)

Reply:
done

Reply:
resolve if ok